op-node/rollup/derive/engine_queue.go

```diff
@@ -221,10 +233,32 @@ func (eq *EngineQueue) SafeL2Head() eth.L2BlockRef {
 	return eq.safeHead
 }
 
+func (eq *EngineQueue) EngineSyncTarget() eth.L2BlockRef {
+	return eq.engineSyncTarget
+}
+
+// Determine if the engine is syncing to the target block
+func (eq *EngineQueue) isEngineSyncing() bool {
+	return eq.unsafeHead.Hash != eq.engineSyncTarget.Hash
+}
+
 func (eq *EngineQueue) Step(ctx context.Context) error {
 	if eq.needForkchoiceUpdate {
 		return eq.tryUpdateEngine(ctx)
 	}
+	// Trying unsafe payload should be done before safe attributes
+	// It allows the unsafe head can move forward while the long-range consolidation is in progress.
+	if eq.unsafePayloads.Len() > 0 {
+		err := eq.tryNextUnsafePayload(ctx)
+		// EOF error means we can't process the next unsafe payload. Then we should process next safe attributes.
+		if err != io.EOF {
```

Comment:
If `eq.syncCfg.EngineP2PEnabled` is true, `tryNextUnsafePayload` never returns `io.EOF` error, so control flow always falls in here. 

So why not 
```suggestion
		if eq.syncCfg.EngineP2PEnabled {
```